src/compiler/commandLineParser.ts

```diff
@@ -405,13 +405,107 @@ namespace ts {
+                if (currentCharCode === CharacterCodes.backslash
+                    && nextCharCode !== undefined) {
+                    // Found an escaped character 
+                    // consume the \ and the escaped char
+                    result += currentChar;
+                    result += jsonText.charAt(i + 1);
+                    i += 1;
+                }
+                else if (currentCharCode === CharacterCodes.doubleQuote) {
+                    // End of string
+                    result += currentChar;
+                    processingString = false;
+                }
+                else {
+                   // String content
+                   result += currentChar;
+                }
+            }
+            else if (processingSingleLineComment) {
+                if (isLineBreak(currentCharCode)) {
+                    // End of single line comment
+                    processingSingleLineComment = false;
+                    // Keep the line breaks to keep line numbers aligned
+                    result += currentChar;
+                }
+                else {
+                   // replace comment content by whitespaces
+                   result += " ";
+                }
+            }
+            else if (processingMultiLineComment) {
+                if (currentCharCode === CharacterCodes.asterisk && nextCharCode === CharacterCodes.slash) {
+                    // End of comment */
+                    result += "  ";
+                    i += 1;
+                    processingMultiLineComment = false;
+                }
+                else if (isLineBreak(currentCharCode)) {
+                    // Keep the line breaks to Keep line aligned
+                    result += currentChar;
+                }
+                else {
+                   // replace comment content by whitespaces
+                   result += " ";
+                }
+            }
+            else if (currentCharCode === CharacterCodes.doubleQuote) {
+                // String start
+                result += currentChar;
+                processingString = true;
+            }
+            else if (currentCharCode === CharacterCodes.hash) {
+                // Start of # comment
+                result += " ";
+                processingSingleLineComment = true;
+            }
+            else if (currentCharCode === CharacterCodes.slash && nextCharCode === CharacterCodes.slash) {
+                // Start of // comment
+                result += "  ";
+                i += 1;
+                processingSingleLineComment = true;
+            }
+            else if (currentCharCode === CharacterCodes.slash && nextCharCode === CharacterCodes.asterisk) {
+                // Start of /**/ comment
+                result += "  ";
+                i += 1;
+                processingMultiLineComment = true;
+            }
+            else {
+                // Keep other characters
+                result += currentChar;

```

Comment:
This implies you're going to be concatenating a string for every character in your `tsconfig.json`. You should instead concatenate the longest sequences of non-comment substrings.

For an example of where we do this, check out `scanString` in `scanner.ts`. We just loop on an index and check the character code at every index. When we hit something like a backslash, we just concatenate a range:

```typescript
function scanString(): string {
    let quote = text.charCodeAt(pos++);
    let result = "";
    let start = pos;
    while (true) {
        if (pos >= end) {
            result += text.substring(start, pos);
            tokenIsUnterminated = true;
            error(Diagnostics.Unterminated_string_literal);
            break;
        }
        let ch = text.charCodeAt(pos);
        if (ch === quote) {
            result += text.substring(start, pos);
            pos++;
            break;
        }
        if (ch === CharacterCodes.backslash) {
            result += text.substring(start, pos);
            result += scanEscapeSequence();
            start = pos;
            continue;
        }
        if (isLineBreak(ch)) {
            result += text.substring(start, pos);
            tokenIsUnterminated = true;
            error(Diagnostics.Unterminated_string_literal);
            break;
        }
        pos++;
    }
    return result;
}
```